### ISSUE TITLE
[Bug]: Use of `window.navigator` breaks SSR usage

### ISSUE THREAD

**Issue by metonym** [carbon-design-system/carbon-charts#1755]
### Application/Team

N/A

### What happened?

I upgraded my [SvelteKit example](https://github.com/metonym/carbon-charts-svelte-examples/tree/master/sveltekit) to use the latest version (1.14.x), but experienced an error related to SSR.

`window.navigator` is a browser API, so it should be guarded against.

Stack trace leads to this line in `configuration.ts`: https://github.com/carbon-design-system/carbon-charts/blob/417dd7b10f946f80c3097c82fb27e3d10782e17d/packages/core/src/configuration.ts#L79

Related PR: https://github.com/carbon-design-system/carbon-charts/pull/1751

### Version

I first started seeing this in 1.14.x

1.13.x works fine.

### Data & options used

```js
See the example repo: https://github.com/metonym/carbon-charts-svelte-examples/tree/master/sveltekit
```

### Relevant log output

```js
9:42:37 AM [vite] Error when evaluating SSR module /node_modules/@carbon/charts-svelte/dist/AlluvialChart.svelte: failed to import "@carbon/charts"
|- ReferenceError: navigator is not defined
    at <redacted>/carbon-charts-svelte-examples/sveltekit/node_modules/@carbon/charts/dist/color-scale-utils-BosGSMyP.mjs:1867:10
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)

9:42:37 AM [vite] Error when evaluating SSR module /node_modules/@carbon/charts-svelte/dist/index.js: failed to import "/node_modules/@carbon/charts-svelte/dist/AlluvialChart.svelte"
|- ReferenceError: navigator is not defined
    at <redacted>/carbon-charts-svelte-examples/sveltekit/node_modules/@carbon/charts/dist/color-scale-utils-BosGSMyP.mjs:1867:10
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)

9:42:37 AM [vite] Error when evaluating SSR module /src/lib/Basic.svelte: failed to import "/node_modules/@carbon/charts-svelte/dist/index.js"
|- ReferenceError: navigator is not defined
    at <redacted>/carbon-charts-svelte-examples/sveltekit/node_modules/@carbon/charts/dist/color-scale-utils-BosGSMyP.mjs:1867:10
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)

9:42:37 AM [vite] Error when evaluating SSR module /src/routes/+page.svelte: failed to import "/src/lib/Basic.svelte"
|- ReferenceError: navigator is not defined
    at <redacted>/carbon-charts-svelte-examples/sveltekit/node_modules/@carbon/charts/dist/color-scale-utils-BosGSMyP.mjs:1867:10
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)

ReferenceError: navigator is not defined
    at <redacted>/carbon-charts-svelte-examples/sveltekit/node_modules/@carbon/charts/dist/color-scale-utils-BosGSMyP.mjs:1867:10
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)
```

### StackBlitz example

https://github.com/metonym/carbon-charts-svelte-examples/tree/master/sveltekit

### What priority level would this be in your opinion?

P4

**Comment by metonym** [comment]
I marked this as a high priority bug as I believe it impacts all and any SSR usage.

**Comment by metonym** [comment]
If you're experiencing this, this is now patched in version 1.14.2.
